### PR TITLE
Traits proposal

### PR DESCRIPTION
**Pull request by darrelmiller**
To improve the process around working on the traits feature, the contents of the Overlay issue have been moved into a markdown file that lives in the proposals folder in the master branch. The goal is to bring this document to the point where it can be copied and pasted into the official spec when finalized.

### CODE REVIEW

**Comment by darrelmiller** [comment]
- Decide on the name.
- Decided : Mixin references should be array of JSON references
- Merging rules as defined here https://github.com/OAI/OpenAPI-Specification/issues/1442#issuecomment-418549259  by @jstoiko

File: proposals/Traits.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+```
+
+and an operation would "apply" the "pageable" mixin with a $mixin property, as follows:
+```
+paths:
+  /foo:
+    get:
+      description: search for foo resources
+      $mixin: 
+        - pagable
```
Comment (darrelmiller) [inline proposals/Traits.md]: This should be a JSON ref, not just a simple string.

**Comment by darrelmiller** [comment]
We need to collect a set of use cases for internal vs external "overlay/trait/mixin/whatever".  @tedepstein  volunteered to help.

**Comment by cmheazel** [comment]
Yes, we will do the same thing with Alternative Schema.

**Comment by cmheazel** [comment]
@darrelmiller where will the use cases live?  Should we have discrete directories for each proposal?  That way we don't have to include use cases and other supporting material in the markup targeted for the OAS spec.

**Comment by earth2marsh** [comment]
Regarding use cases, there were a few from #1442 I'll bring over here:
as a way to keep any implementation details for a spec se

1. parate from the consumer-oriented doc, such as API management concerns from the API provider's perspective, for example attaching a policy to an operation.
2. to allow a writer to create better descriptions/summaries/titles for attributes like a parameter or an operation without having to edit the spec itself. But since parameters are an array, in order to edit a single description, I would have to reproduce the entire array, no? Is there a way to more precisely target?
3. To identify an SLA. You might have a valid spec that doesn't have that information, but you could ask for the SLA variant for that spec.
4. Codegen is another, as we're polluting specs with metadata for codegen. This gives us a way to separate that out.

**Comment by darrelmiller** [comment]
@cmheazel I was considering the use cases as living in the comments here, but I see your point that it is going to get just as messy.  Perhaps each proposal should have a folder and the spec wording and all related documents, e.g. examples, use cases, spec wording should all live in that folder.

Having a permenant record of why we created a feature is definitely not a bad thing.

**Comment by MikeRalphson** [comment]
As a result of #2604 the filename for this proposal should change to use a dated prefix. Please could you update this PR to match? Apologies for the extra work.

**Pull request closed without merging**